### PR TITLE
Add HSM API http client

### PR DESCRIPTION
**Pull request by skos001**

### CODE REVIEW

File: src/Crypto/Signing/IHsmSignerService.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+// Licensed under the EUROPEAN UNION PUBLIC LICENCE v. 1.2
+// SPDX-License-Identifier: EUPL-1.2
+
+using System.Threading.Tasks;
+
+namespace NL.Rijksoverheid.ExposureNotification.BackEnd.Crypto.Signing;
+
+public interface IHsmSignerService
+{
+    Task<byte[]> GetNlSignatureAsync(byte[] content);
```
Comment (hvisser) [inline src/Crypto/Signing/IHsmSignerService.cs:11]: nit: maybe rename this to `GetCmsSignatureAsync`?

File: src/Crypto/Signing/HsmSignerService.cs [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+            _httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+        }
+
+        public async Task<byte[]> GetNlSignatureAsync(byte[] content)
+        {
+            using var sha256Hasher = SHA256.Create();
+            var contentHash = sha256Hasher.ComputeHash(content);
+            var contentHashBased64 = Convert.ToBase64String(contentHash);
+
+            var certificateChain = GetCertificateChain();
```
Comment (hvisser) [inline src/Crypto/Signing/HsmSignerService.cs:44]: I think we can get rid of this and just store the entire chain in nlPem here...If that's part of the clean up, then just ignore me 😄

File: src/Crypto/Signing/HsmSignerService.cs [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+            _httpClient.DefaultRequestHeaders.Remove("Authorization");
+            _httpClient.DefaultRequestHeaders.Add("Authorization", _config.NlJwt);
+
+            var response = _httpClient.PostAsJsonAsync("/cms", requestModel).Result;
+
+            if (!response.IsSuccessStatusCode)
+            {
+                //TODO: error handling
+                return Array.Empty<byte>();
```
Comment (hvisser) [inline src/Crypto/Signing/HsmSignerService.cs:79]: Could we just throw here? To prevent invalid signatures?

**Review by hvisser — APPROVED** [review]

File: src/Crypto/Signing/IHsmSignerService.cs (hunk 1 above)
Comment (skos001) [reply src/Crypto/Signing/IHsmSignerService.cs]: Good point, done (in more places than just here though... 😅 )

File: src/Crypto/Signing/HsmSignerService.cs (hunk 2 above)
Comment (skos001) [reply src/Crypto/Signing/HsmSignerService.cs]: Indeed you are correct... removed the offending code 🥳

File: src/Crypto/Signing/HsmSignerService.cs (hunk 3 above)
Comment (skos001) [reply src/Crypto/Signing/HsmSignerService.cs]: Yeah, good point. Throwing some semi-random exceptions for now, so we can get testing 😅

File: src/Crypto/Signing/HsmSignerService.cs [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+            _httpClient.DefaultRequestHeaders.Remove("Authorization");
+            _httpClient.DefaultRequestHeaders.Add("Authorization", _config.CmsJwt);
+
+            var response = _httpClient.PostAsJsonAsync("/cms", requestModel).Result;
+
+            if (!response.IsSuccessStatusCode)
+            {
+                //TODO: improve error handling
+                _logger.LogError("HTTP request to cms endpoint failed");
+                throw new Exception();
```
Comment (hvisser) [inline src/Crypto/Signing/HsmSignerService.cs:67]: nit: does it make sense to add the http response code to both the logging and the exception here? Or maybe whatever the response produced it's not successful? When adding it both to the exception and the logging then I'd hope you'd always see the error even if logging is misconfigured / turned off.

Comment (skos001) [reply src/Crypto/Signing/HsmSignerService.cs:67]: You're not wrong 😉 I've at least added the status code to the message being logged 👍

**Pull request merged**